src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -17,30 +17,34 @@ import {requestExternalStoragePermission} from '../../../utils/permission';
 import {setChannel} from '../../../context/actions/setChannel';
 import {setParticipants} from '../../../context/actions/groupChat';
 
-const useGroupInfo = () => {
-  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
+const useGroupInfo = (channelId = null) => {
   const navigation = useNavigation();
-  const {participants, asset} = groupChatState;
+
+  const [groupChatState, groupPatchDispatch] = React.useContext(Context).groupChat;
+  const [, dispatchChannel] = React.useContext(Context).channel;
   const [client] = React.useContext(Context).client;
   const [channelState] = React.useContext(Context).channel;
   const [profile] = React.useContext(Context).profile;
-  const {channel, profileChannel} = channelState;
+
+  const blockModalRef = React.useRef(null);
+
   const [isLoadingMembers, setIsLoadingMembers] = React.useState(false);
   const [uploadedImage, setUploadedImage] = React.useState('');
   const [isUploadingImage, setIsUploadingImage] = React.useState(false);
   const [isLoadingInitChat, setIsLoadingInitChat] = React.useState(false);
   const [username, setUsername] = React.useState(channelState.channel?.data?.name);
-  const createChat = channelState.channel?.data?.created_at;
-  const countUser = Object.entries(participants).length;
   const [selectedUser, setSelectedUser] = React.useState(null);
   const [newParticipant, setNewParticipant] = React.useState([]);
   const [openModal, setOpenModal] = React.useState(false);
   const [isAnonymousModalOpen, setIsAnonymousModalOpen] = React.useState(false);
   const [isFetchingAllowAnonDM, setIsFetchingAllowAnonDM] = React.useState(false);
-  const [, dispatchChannel] = React.useContext(Context).channel;
+
   const {createSignChat, handleAnonymousMessage} = useCreateChat();
-  const blockModalRef = React.useRef(null);
 
+  const {participants, asset} = groupChatState;
+  const countUser = Object.entries(participants).length;
+  const {channel, profileChannel} = channelState;
+  const createChat = channelState.channel?.data?.created_at;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useGroupInfo` hook now takes an optional `channelId` parameter. This is a good change as it allows the hook to be more flexible and reusable. However, I noticed that you're using `channelState.channel?.data?.name` to set the initial state of `username`. If `channelId` is provided, it might not match with the current `channelState`, leading to incorrect username being set initially. Consider fetching the channel data based on the provided `channelId` to ensure accuracy.
```diff
- const [username, setUsername] = React.useState(channelState.channel?.data?.name);
+ const [username, setUsername] = React.useState(null);

+ React.useEffect(() => {
+   if (channelId) {
+     // Fetch channel data based on channelId and set the username
+   } else {
+     setUsername(channelState.channel?.data?.name);
+   }
+ }, [channelId, channelState]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->